api/desecapi/serializers.py

```diff
@@ -681,7 +681,8 @@ def to_internal_value(self, data):
         except KeyError:
             raise serializers.ValidationError({'code': ['This field is required.']})
         except ValueError:
-            validity = self.context['view'].validity_period or 'unlimited'
+            if (validity := self.context['view'].validity_period) is None:
+                validity = 'unlimited'
```

Comment:
The previous version (`validity = self.context['view'].validity_period or 'unlimited'`) would give `'unlimited'` if e.g. `validity_period == datetime.timedelta(hours=0)`, which was not intended. I don't like this solution, but it's the least excessive one I could come up with. What do you think, @nils-wisiol ?

Reply:
How about this?

```python
validity = 'unlimited' if (_ := self.context['view'].validity_period) is None else _
```

Looks very PHP-ey ...

Reply:
maybe the following is best:

```python
validity = self.context['view'].validity_period
if validity is None:
    validity = 'unlimited'
```

Reply:
validity = 'unlimited' if (_ := self.context['view'].validity_period) is None else _

omg I didn't know this was possible 😳

Let's stick with the current version. Explicit it better. (But also this comes to mind: `getattr(self.context['view'], 'validity_period', default=None`)